### ISSUE TITLE
Add support for Google Assistant

### ISSUE THREAD

**Issue by blackwind** [vanilla-music/vanilla#705]
It'd be great if, when I'm at the gym, I could simply say through my headset, "OK Google, play Bon Jovi on Vanilla Music", instead of constantly breaking up my workout to scroll through my library whenever I want something new. Google has a blog about how this might be accomplished [here](https://android-developers.googleblog.com/2017/10/the-google-assistant-can-help-control.html), including the following sample image of integration with a different app:

![Sample](https://1.bp.blogspot.com/-PTdKrTezb1c/WdvfTZY5thI/AAAAAAAAEpo/CzDPTN6gCfk0b9MksA96QceL8WLy5vi2ACLcBGAs/s1600/image1.gif)

Currently, Assistant can do things like pausing and skipping tracks when Vanilla is already playing, but asking it to jump to particular artists or songs just brings up a window asking me if I want to do this in the YouTube app or the Play Store app.

**Comment by adrian-bl** [comment]
We already do support MediaSessions and even have a MediaBrowser implementation - maybe we are missing some callbacks?

https://developer.android.com/guide/topics/media-apps/interacting-with-assistant.html#use_a_media_session (like: onPlayFromSearch)

**Comment by adrian-bl** [comment]
https://github.com/vanilla-music/vanilla/commit/b82dc3cab180d0f19901eb74aa8ba3b75ec123aa seems to be enough to at least get triggered via the old 'ok google' thingy.

Doesn't do much yet as i'm not sure how to get the actual search result (i get the user search string, but it would be nice to know whether google thinks its a song or an album).
Not sure how well this works in Assistant.

**Comment by Adonai** [comment]
> but it would be nice to know whether google thinks its a song or an album

![Why not both?](http://i0.kym-cdn.com/photos/images/newsfeed/000/538/731/0fc.gif)
I think we can get things done with some fuzzy search impl, no?

**Comment by adrian-bl** [comment]
Yes, that would work. But at least the voice search app knows wich part of the query is the artist and which one is the album or song. Would be nice to get this info, but looks like it is not passed to us.

Also wendo not show up in the assistant - yet.

**Comment by adrian-bl** [comment]
The latest nightly now works with voice search. Assistant support is still missing.
